pkg/tkctl/cmd/check/check.go

```diff
@@ -0,0 +1,460 @@
+			}
+		case "tikv-configmap.yaml":
+			if err := yaml.Unmarshal([]byte(v), &tikvConfigMap); err != nil {
+				return err
+			}
+		case "pd-configmap.yaml":
+			if err := yaml.Unmarshal([]byte(v), &pdConfigMap); err != nil {
+				return err
+			}
+		case "pump-configmap-rollout.yaml":
+			if err := yaml.Unmarshal([]byte(v), &pumpConfigMap); err != nil {
+				return err
+			}
+		case "drainer-configmap-rollout.yaml":
+			if err := yaml.Unmarshal([]byte(v), &drainerConfigMap); err != nil {
+				return err
+			}
+		case "tidb-service.yaml":
+			if err := yaml.Unmarshal([]byte(v), &tidbSvc); err != nil {
+				return err
+			}
+		case "pump-statefulset.yaml":
+			if err := yaml.Unmarshal([]byte(v), &pumpSts); err != nil {
+				return err
+			}
+		case "drainer-statefulset.yaml":
+			if err := yaml.Unmarshal([]byte(v), &drainerSts); err != nil {
+				return err
+			}
+		}
+	}
+	if !componentVersionMatches(&tidbcluster, pumpSts, drainerSts) {
+		errorPrintln("component version mismatch")
+		return errors.New("cluster component version mismatch")
+	}
+	infoPrintln("component version matches, all versions are ", getImageVersion(tidbcluster.Spec.PD.Image))
+
+	o.checkTidbService(tidbSvc)
+	checkResourceSettings(&tidbcluster, pumpSts, drainerSts)
+
+	var tidbcfg *config_v2_1_16.TidbConfig
+	var pdcfg *config_v2_1_16.PdConfig
+	var tikvcfg *config_v2_1_16.TikvConfig
+	if cfg, exist := tidbConfigMap.Data["config-file"]; exist {
+		infoPrintln("#### TiDB Config ####")
+		fmt.Println(cfg)
+		infoPrintln("#### TiDB Config ####\n")
+		_, err = toml.Decode(cfg, &tidbcfg)
+		if err != nil {
+			return fmt.Errorf("failed to decode tidb config: %v", err)
+		}
+		if tidbcfg.Log.Level != "info" {
+			warnPrintln("TiDB log-level:", tidbcfg.Log.Level)
+		}
+		if tidbcfg.Binlog.IgnoreError {
+			warnPrintln("TiDB binlog.ignore-error is enabled")
+		}
+		if tidbcfg.PreparedPlanCache.Enabled {
+			warnPrintln("TiDB prepared-plan-cache is experimental feature")
+		}
+		if tidbcfg.TxnLocalLatches.Enabled {
+			warnPrintln("TiDB txn-local-latches is for write conflict heavy business")
+		}
+	}
+	if cfg, exist := tikvConfigMap.Data["config-file"]; exist {
+		infoPrintln("#### TiKV Config ####")
+		fmt.Println(cfg)
+		infoPrintln("#### TiKV Config ####\n")
+		_, err = toml.Decode(cfg, &tikvcfg)
+		if err != nil {
+			return fmt.Errorf("failed to decode tikv config: %v", err)
+		}
+		tikvResources := util.ResourceRequirement(tidbcluster.Spec.TiKV.ContainerSpec)
+		memoryLimit := tikvResources.Limits[corev1.ResourceMemory]
+		m, b1 := memoryLimit.AsInt64()
+		if b1 {
+			// rocksdb.writecf.block-cache-size 10%~30% of memory limit
+			// rocksdb.defaultcf.block-cache-size 30%~50% of memory limit
+			if int64(tikvcfg.Rocksdb.Writecf.BlockCacheSize) < m/10 || int64(tikvcfg.Rocksdb.Writecf.BlockCacheSize) > 3*m/10 {
+				errorPrintf("TiKV rocksdb.writecf.block-cache-size should be 10%% ~ 30%% of memory limit, but got %d/%d\n", tikvcfg.Rocksdb.Writecf.BlockCacheSize, m)
```

Comment:
Now TiKV configuration uses a separate type (`ReadableSize`) for `size`, it will fail to parse `KiB`, `MiB`, `GiB` and `TiB`. This is for `2.1.16`, for `3.0.0` and later, we'll use `ByteSize` just like PD.